### PR TITLE
Adding Basic Tests for Controllers 

### PR DESCRIPTION
**Pull request by MarcoDDM**
This PR adds essential controller tests:

- For `UserPostsController `and `UsersController`.
- Verifying successful responses.
- Checking template rendering.
- Validating placeholder text in responses.
- Added screenshot for code reviewer's convenience:
rspec_controllers

**Tests Cover:**

_UserPostsController:_
- GET #index
- GET #show

_UsersController:_
- GET #index
- GET #show

Thanks for your time

### CODE REVIEW

File: spec/requests/users_spec.rb [hunk 1]
```diff
@@ -1,7 +1,25 @@
...
+      expect(response).to have_http_status(200)
+    end
+
+    it 'includes the correct placeholder text in the response body' do
+      puts 'Expecting to receive the correct placeholder text for users show...'
+      get '/users/:id'
+      expect(response.body).to include('User Profile')
+    end
   end
 end
```
Comment (Munsa1) [inline spec/requests/users_spec.rb]: - [x] It's a requirement that your Spec files include tests that ensure the correct view template is rendered.
Example : 
`expect(response).to render_template(:show)`


#### Note that this should be done for both the Index and Show views of both the User and Post view files.

File: spec/requests/user_posts_spec.rb [hunk 2]
```diff
@@ -1,7 +1,33 @@
...
+    end
+  end
+
+  describe 'GET #show' do
+    it 'returns a successful response' do
+      get user_path(id: 1)
+      expect(response).to have_http_status(:success)
+    end
   end
 end
```
Comment (Munsa1) [inline spec/requests/user_posts_spec.rb]: - [x] It's a requirement that you include tests that ensure the response body includes the correct placeholder text. Example : `expect(response.body).to include("Welcome to user index page")`

Kindly do this for the index and show views for the Post as well

File: spec/requests/users_spec.rb [hunk 3]
```diff
@@ -1,7 +1,25 @@
...
-  describe 'GET /index' do
-    pending "add some examples (or delete) #{__FILE__}"
+RSpec.describe UsersController, type: :request do
+  describe 'Request spec for users index' do
+    it 'returns a successful response' do
+      puts 'Checking HTTP status for users index...'
+      get '/users'
+      expect(response).to have_http_status(200)
+    end
+  end
```
Comment (Munsa1) [inline spec/requests/users_spec.rb]: - [x] Kindly add a test that ensures the response body includes the correct placeholder text for the index page of the User. Example : `expect(response.body).to include("Welcome to user index page")`

File: spec/requests/user_posts_spec.rb (hunk 2 above)
Comment (Munsa1) [inline spec/requests/user_posts_spec.rb]: - [x] Again here remember to add tests that ensure the correct view template is rendered for both the index and show pages of the Post views

e.g
`expect(response).to render_template(:show)`

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @MarcoDDM, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
Passing Test Cases  :heavy_check_mark: 
Correct Git Flow :heavy_check_mark: 
Descriptive ReadMe File :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Reem-lab — APPROVED** [review]
# Hi @MarcoDDM,  👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights

- [ ]  Followed Gitflow 💯
- [ ]  Descriptive PR⭐
- [ ]  Kudos for working on the tests🤩
- [ ]  All tests passed  🎉
- [ ] Good job working on the requested changes 🙏🏻


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**